scertec.go

```diff
@@ -64,32 +66,36 @@ func NewClient(ctx context.Context, c setec.Client, cache setec.Cache, prefix st
 	if len(domains) == 0 {
 		return nil, errors.New("no domains provided")
 	}
-	var secretNames []string
+	allowed := make(map[string]bool, len(domains))
 	for _, d := range domains {
-		secretNames = append(secretNames,
-			secretName(prefix, d, "rsa"),
-			secretName(prefix, d, "ecdsa"),
-		)
+		allowed[d] = true
 	}
 	st, err := setec.NewStore(ctx, setec.StoreConfig{
-		Client:  c,
-		Secrets: secretNames,
-		Cache:   cache,
+		Client:      c,
+		Cache:       cache,
```

Comment:
Given we already have a cache, another possible approach would be to prime the cache by hand and then use declared secrets so you don't have to keep the allowlist yourself, but this is fine too.

Reply:
Prime the cache with what, though? I don't want to block startup, so I don't want to go reading the values from setec.

Reply:
Even if you don't want to do a manual startup to prime the cache, you can curl the values out and bash them into JSON with jq. The format it writes to the cache is documented and I don't plan to change it in incompatible ways. (That said, I mention this mainly as another way to get the effect)